search/domain/base.py

```diff
@@ -257,87 +183,3 @@ class SimpleQuery(Query):
 
     include_cross_list: bool = field(default=True)
     """If True, secondaries are considered when limiting by classification."""
-
-
-@dataclass(init=True)
-class Document:
-    """A search document, representing an arXiv paper."""
-
-    submitted_date: Optional[datetime] = None
-    announced_date_first: Optional[date] = None
-    submitted_date_first: Optional[datetime] = None
-    submitted_date_latest: Optional[datetime] = None
-    submitted_date_all: List[str] = field(default_factory=list)
-    id: str = field(default_factory=str)
-    abstract: str = field(default_factory=str)
-    abstract_tex: str = field(default_factory=str)
-    authors: List[Person] = field(default_factory=list)
-    authors_freeform: str = field(default_factory=str)
-    owners: List[Person] = field(default_factory=list)
-    modified_date: str = field(default_factory=str)
-    updated_date: str = field(default_factory=str)
-    is_current: bool = True
-    is_withdrawn: bool = False
-    license: Dict[str, str] = field(default_factory=dict)
-    paper_id: str = field(default_factory=str)
-    paper_id_v: str = field(default_factory=str)
-    title: str = field(default_factory=str)
-    title_tex: str = field(default_factory=str)
-    source: Dict[str, Any] = field(default_factory=dict)
-    version: int = 1
-    latest: str = field(default_factory=str)
-    latest_version: int = 0
-    submitter: Optional[Person] = field(default=None)
-    report_num: str = field(default_factory=str)
-    proxy: bool = False
-    msc_class: List[str] = field(default_factory=list)
-    acm_class: List[str] = field(default_factory=list)
-    metadata_id: int = -1
-    journal_ref: str = field(default_factory=str)
-    doi: str = field(default_factory=str)
-    comments: str = field(default_factory=str)
-    abs_categories: str = field(default_factory=str)
-    formats: List[str] = field(default_factory=list)
-    primary_classification: Classification = field(
-        default_factory=Classification
-    )
-    secondary_classification: ClassificationList = field(
-        default_factory=ClassificationList
-    )
-
-    score: float = 1.0
```

Comment:
We lose the defaults when moving to `TypedDict`. Flagging just in case.

Reply:
Yeah, that was one of the things that I worried about most. Will take another pass, but I believe that we aren't relying on that being set

Reply:
Indeed, not used currently